### PR TITLE
[quant] Added relu6 kernel

### PR DESCRIPTION
**Pull request by z-a-f**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#24799 [quant] Added relu6 kernel**

Pull Request resolved: https://github.com/pytorch/pytorch/pull/24799

Differential Revision: [D16875493](https://our.internmc.facebook.com/intern/diff/D16875493)

### CODE REVIEW

File: torch/nn/quantized/modules/activation.py [hunk 1]
```diff
@@ -39,3 +41,38 @@ def forward(self, input):
     @staticmethod
     def from_float(mod):
         return ReLU(mod.inplace)
+
+
+class ReLU6(NNReLU):
```
Comment (jerryzh168) [inline torch/nn/quantized/modules/activation.py]: nit: let's use nn.ReLU to be more explict. same for the previous one.

File: aten/src/ATen/native/quantized/cpu/qrelu.cpp [hunk 2]
```diff
@@ -3,6 +3,7 @@
 #include <ATen/core/op_registration/op_registration.h>
 #include <ATen/native/TensorIterator.h>
 #include <ATen/native/cpu/Loops.h>
+#include <ATen/quantized/Quantizer.h>
```
Comment (jerryzh168) [inline aten/src/ATen/native/quantized/cpu/qrelu.cpp]: unused?

**Review by jerryzh168 — APPROVED** [review]
LG, some nit comments

File: torch/nn/quantized/modules/activation.py [hunk 3]
```diff
@@ -39,3 +41,38 @@ def forward(self, input):
...
+    Examples::
+
+        >>> m = nn.quantized.ReLU6()
+        >>> input = torch.randn(2)
+        >>> input = torch.quantize_linear(input, 1.0, 0, dtype=torch.qint32)
+        >>> output = m(input)
+    """
+    def __init__(self, inplace=False):
+        super(ReLU6, self).__init__(inplace)
+        assert not inplace, 'torch.nn.quantized.ReLU does not support inplace'
```
Comment (llyfacebook) [inline torch/nn/quantized/modules/activation.py]: Need inplace, :)

File: torch/nn/quantized/modules/activation.py [hunk 4]
```diff
@@ -39,3 +41,38 @@ def forward(self, input):
...
+
+    def forward(self, input):
+        return torch.ops.quantized.relu6(input)
+
+    @staticmethod
+    def from_float(mod):
+        return ReLU(mod.inplace)
```
Comment (llyfacebook) [inline torch/nn/quantized/modules/activation.py]: ReLU6(mod.inplace)?

File: torch/nn/quantized/modules/activation.py (hunk 3 above)
Comment (zafartahirov) [reply torch/nn/quantized/modules/activation.py]: Inplace is part of another diff

File: aten/src/ATen/native/quantized/cpu/qrelu.cpp (hunk 2 above)
Comment (zafartahirov) [reply aten/src/ATen/native/quantized/cpu/qrelu.cpp]: I think I need it for the `at::quantize_val`.

**Pull request closed without merging**